src/dodal/beamlines/p38.py

```diff
@@ -312,7 +313,7 @@ def linkam(
     return device_instantiation(
         Linkam3,
         "linkam",
-        "-EA-LINKM-02:",
+        f"{BeamlinePrefix(BL).insertion_prefix}-EA-LINKM-02:",
```

Comment:
While device_factory is reverted, this isn't going to work and is unrelated: it'll cause merge issues when that change goes back in, so let's revert it for now.

```suggestion
        "-EA-LINKM-02:",
```